p2p/src/testing_utils.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use libp2p::Multiaddr;
+use std::net::SocketAddr;
+
+/// An interface for creating the address.
+///
+/// This abstraction layer is needed to uniformly create an address in the tests for different
+/// mocks transport implementations.
+pub trait MakeTestAddress {
```

Comment:
I think this trait should have a different name now. We should call it something like "TestTransport". The docs should be updated too.